lib/core/ingestion/es_sink.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'app/config'
+require 'utility/bulk_queue'
+require 'utility/es_client'
+require 'utility/logger'
+require 'elasticsearch/api'
+
+module Core
+  module Ingestion
+    class EsSink
+      def initialize(index_name, request_pipeline, bulk_queue = Utility::BulkQueue.new)
+        @client = Utility::EsClient.new(App::Config[:elasticsearch])
+        @index_name = index_name
+        @request_pipeline = request_pipeline
+        @operation_queue = bulk_queue
+      end
+
+      def ingest(id, serialized_document)
+        index_op = serialize({ 'index' => { '_index' => index_name, '_id' => id } })
+
+        flush unless @operation_queue.will_fit?(index_op, serialized_document)
+
+        @operation_queue.add(
```

Comment:
If the size of a single doc is over the size threshold, this will throw an overflow error

Reply:
I think oversized doc should be discarded.